### PR TITLE
Fix issue #628 with '.A' stuff on csr_matrix

### LINKED ISSUE #629: [BUG] Attribute Error with csr_matrix using Recommender based models
### Description
Using any of the model based on Recommender abstract class causes an Attribute Error Exception: csr_matrix object has no attribute A.

Tested with ItemKNN and UserKnn so far. SVD model works well with same dataset.

### In which platform does it happen?
MacOS 14.5 with arm platform (M1)
Python3.12

### How do we replicate the issue?
MWE:
```python
from cornac.data import Reader, Dataset
from cornac.eval_methods import StratifiedSplit
from cornac.models import ItemKNN, UserKNN, SVD
from cornac.datasets import movielens
data = movielens.load_feedback(variant="1M")
dataset= Dataset.from_uir(data)
split= StratifiedSplit(data, test_size=0.2, seed=42, rating_threshold=4)
train_data=split.train_set
test_set=split.test_set

item_knn = ItemKNN(k=10)
item_knn.fit(train_data)

user_knn = UserKNN(k=10)
user_knn.fit(train_data)

svd = SVD()
svd.fit(train_data)

item_knn.recommend(user_id='42', k=5) <--- issue from car_matrix
user_knn.recommend(user_id='42', k=5) <--- same
svd.recommend(user_id='42', k=5) <----- no issue here
```

Issue traceback seems to indicate an issue in the source code, where ui_mat is defined with csr_matrix, and using ui_mat.A.ravel() in file `cornac/models/knn/recom_knn.py`

### Expected behavior (i.e. solution)

Expect the models to recommend as the SVD does, without throwing any internal error.

### PR DESCRIPTION
**Pull request by quentinhaenn**
### Description

Fix the issue concerning '.A' alias deprecated with latest Scipy version (from 1.11).
Change '.A' to '.toarray()' in scoring functions in knn models.
Also fixing upper bound on numpy version until compatibility is done in requirements

### Related Issues

Fix #629 

### Checklist:

- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated `README.md` (if you are adding a new model).
- [ ] I have updated `examples/README.md` (if you are adding a new example).
- [ ] I have updated `datasets/README.md` (if you are adding a new dataset).

### CODE REVIEW

File: cornac/data/text.py [hunk 1]
```diff
@@ -974,6 +974,7 @@ def batch_tfidf(self, batch_ids, keep_sparse=False):
         tfidf_mat = self.tfidf_matrix[batch_ids]
         return tfidf_mat if keep_sparse else tfidf_mat.A
```
Comment (tqtg) [inline cornac/data/text.py:975]: looks like we need to change this

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-numpy
+numpy<=1.26
```
Comment (tqtg) [inline requirements.txt:1]: let's make it `numpy<2.0`

**Review by tqtg — CHANGES_REQUESTED** [review]

File: cornac/data/text.py (hunk 1 above)
Comment (quentinhaenn) [reply cornac/data/text.py]: Yeah I missed it. Also modifying the same modification in same file, l.954 on bow_mat.

File: requirements.txt (hunk 2 above)
Comment (quentinhaenn) [reply requirements.txt]: Done in last commit

**Review by tqtg — APPROVED** [review]

**Pull request merged**